src/main/java/fr/insalyon/creatis/gasw/plugin/executor/dirac/execution/DiracMonitor.java

```diff
@@ -174,23 +215,39 @@ public void run() {
                         logger.error(cout);
                     }
                     closeProcess(process);
+                    checkMissingDiracJob(command.subList(1, command.size()), jobIdsReturnedByDirac);
                 }
                 Thread.sleep(GaswConfiguration.getInstance().getDefaultSleeptime());
-            } catch (IOException ex) {
-                logger.error(ex);
-            } catch (GaswException ex) {
-                // do nothing
-            } catch (DAOException ex) {
-                // do nothing
+            } catch (IOException | GaswException | DAOException ex) {
+                logger.error("[DIRAC] error monitoring DIRAC jobs", ex);
             } catch (InterruptedException ex) {
-                logger.error(ex);
+                logger.error("[DIRAC] jobs monitoring thread interrupted" + ex);
             } finally {
                 closeProcess(process);
             }
         }
 
     }
 
+    private void checkMissingDiracJob(List<String> sentJobIds, List<String> returnedJobIds) {
+        if (sentJobIds.size() != returnedJobIds.size()) {
+            logger.info("[Dirac] size difference between sent (" + sentJobIds.size() +
+                    ") and received (" + returnedJobIds.size() + ") jobs" +
+                    " when checking status");
+        }
+        StringBuilder missingJobIds = new StringBuilder();
+        for (String sentJobId : sentJobIds) {
+            if ( !returnedJobIds.contains(sentJobId)) {
+                missingJobIds.append(sentJobId);
+                missingJobIds.append(" ");
+            }
+        }
```

Comment:
As discussed, in order to simplify this, we could only retrieve the missingJobIds if the sizes are not equal